docs/Examples/Form.example.purs

```diff
@@ -7,200 +7,155 @@ import Data.Array as Array
-
-      SetUser update ->
-        let
-          formProps =
-            { loadColor: loadColor self.state.simulatePauses
-            , loadColors: loadColors self.state.simulatePauses
-            , readonly: self.state.readonly
-            }
-        in
-          self.setState \s -> s
-            { result = F.revalidate userForm formProps (update s.user)
-            , user = update s.user
-            }
+  component "FormExample" \_ -> React.do
+    { formData, form } <- F.useForm metaForm
+        { initialState: formDefaults
+        , readonly: false
+        , inlineTable: false
+        , forceTopLabels: false
+        }
 
-      Submit ->
-        self.setState \state -> state { user = F.setModified state.user, modalOpen = isJust state.result }
+    pure $ column_
+      [ column
+          { style: css { maxWidth: "50rem", padding: "2rem 0" }
+          , children: [ form ]
+          }
 
-      Reset ->
-        self.setState (const initialState)
+      , example $ element userFormExample formData
+      ]
 
-    render self@{ state: { user, result, inlineTable, forceTopLabels, readonly, simulatePauses, modalOpen } } =
-        column_
-          [ h1_ "Form"
+-- | This form renders the toggles at the top of the example
+metaForm
+  :: forall props
+   . FormBuilder
+      { readonly :: Boolean
+      | props
+      }
+      { inlineTable :: Boolean
+      , forceTopLabels :: Boolean
+      , readonly :: Boolean
+      , simulatePauses :: Boolean
+      }
+      Unit
+metaForm = ado
+  inlineTable <-
+    F.indent "Inline table" Neither
+    $ F.focus (prop (SProxy :: SProxy "inlineTable"))
+    $ F.switch
+  forceTopLabels <-
+    F.indent "Force top labels" Neither
+    $ F.focus (prop (SProxy :: SProxy "forceTopLabels"))
+    $ F.switch
+  readonly <-
+    F.indent "Readonly" Neither
+    $ F.focus (prop (SProxy :: SProxy "readonly"))
+    $ F.switch
+  simulatePauses <-
+    F.indent "Simulate pauses" Neither
+    $ F.focus (prop (SProxy :: SProxy "simulatePauses"))
+    $ F.switch
+  in unit
+
+mkUserFormExample
+  :: CreateComponent
+       { inlineTable :: Boolean
+       , forceTopLabels :: Boolean
+       , readonly :: Boolean
+       , simulatePauses :: Boolean
+       }
+mkUserFormExample = do
+  component "UserFormExample" \props -> React.do
+    modalOpen /\ setModalOpen <- useState false
+
+    { setModified, reset, validated, form } <- F.useForm userForm
+        { initialState: userFormDefaults
+        , readonly: props.readonly
+        , inlineTable: props.inlineTable
+        , forceTopLabels: props.forceTopLabels && not props.inlineTable
+        , loadColor: loadColor props.simulatePauses
+        , loadColors: loadColors props.simulatePauses
+        }
 
-          , column
-              { style: css { maxWidth: "50rem", padding: "2rem 0" }
+    let hasResult = isJust validated
+    useEffect hasResult do
+      setModalOpen $ const hasResult
+      mempty
```

Comment:
Is this `mempty` necessary?

Reply:
Yeah, it's a `pure (pure unit)`

Reply:
Ah, I see; `useEffect` needs to return the canceller, right?

Reply:
Yep